docs/commands.rst

```diff
@@ -0,0 +1,213 @@
+    implemented with just a "pass" statement. The reason being that these
+    commands do not directly send their payload to a node, but rather call
+    core commands that do have their ``RequestFilter`` implemented.
+    More about this topic in the next section.
+
+
+Extended Commands
+-----------------
+
+Core commands, like :py:meth:`~Iota.find_transactions` in the example above,
+are for direct communication with the node for simple tasks such
+as finding a transaction on the Tangle or getting info about the node.
+Extended commands (that serve :ref:`Extended API Methods`) on the other hand
+carry out more complex operations such as combining core commands, building
+objects, etc...
+
+As a consequence, extended commands override the default execution phase of their
+base class.
+
+Observe for example :py:class:`FindTransactionObjectsCommand` extended command
+that is called in :py:meth:`~Iota.find_transaction_objects` extended API
+method. It overrides the :py:meth:`_execute` method of its base class.
+
+Let's take a closer look on the implementation::
+
+    ...
+    def _execute(self, request):
+        bundles = request\
+            .get('bundles')  # type: Optional[Iterable[BundleHash]]
+        addresses = request\
+            .get('addresses')  # type: Optional[Iterable[Address]]
+        tags = request\
+            .get('tags')  # type: Optional[Iterable[Tag]]
+        approvees = request\
+            .get('approvees')  # type: Optional[Iterable[TransactionHash]]
+
+        ft_response = FindTransactionsCommand(adapter=self.adapter)(
+            bundles=bundles,
+            addresses=addresses,
+            tags=tags,
+            approvees=approvees,
+        )
+
+        hashes = ft_response['hashes']
+        transactions = []
+        if hashes:
+            gt_response = GetTrytesCommand(adapter=self.adapter)(hashes=hashes)
+
+            transactions = list(map(
+                Transaction.from_tryte_string,
+                gt_response.get('trytes') or [],
+            ))  # type: List[Transaction]
+
+        return {
+            'transactions': transactions,
+        }
+    ...
+
+Instead of sending the request to the adapter,
+:py:meth:`FindTransactionObjectsCommand._execute` calls
+:py:meth:`FindTransactionsCommand` core command, gathers the transaction hashes
+that it found, and collects the trytes of those transactions by calling
+:py:meth:`GetTrytesCommand` core command. Finally, using the obtained trytes, it constructs
+a list of transaction objects that are returned to
+:py:meth:`~Iota.find_transaction_objects`.
+
+.. important::
+    If you come up with a new functionality for the PyOTA API, please raise
+    an issue in the `PyOTA Bug Tracker`_ to facilitate discussion.
+
+    Once the community agrees on your proposal, you may start implementing
+    a new extended API method and the corresponding extended PyOTA command.
+
+    Contributions are always welcome! :)
```

Comment:
❔ Include a link to https://github.com/iotaledger/iota.py/blob/master/CONTRIBUTING.rst ?